### PR TITLE
Library version updates

### PR DESCRIPTION
**Pull request by MykhailoNester**
[Description]
- remove deprecated kotlin-android-extensions;
- update: gradle / gradle-wrapper / junit / workmanager/ material / converter-gson / gson / logging-interceptor / kotlin-std
- code cleanup;

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ You have two ways to install Crowdin Android SDK.
 
    ```groovy
    dependencies {
-       implementation 'com.github.crowdin.mobile-sdk-android:sdk:1.4.1'
+       implementation 'com.github.crowdin.mobile-sdk-android:sdk:1.4.3'
```
Comment (andrii-bodnar) [inline README.md:68]: @MykhailoNester why 1.4.3?

File: README.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ You have two ways to install Crowdin Android SDK.
    To resolve, either exclude `gson` from Crowdin or from your library is OK, but be sure to keep the newer one for backward-compatibility.
 
    ```groovy
-   implementation ('com.github.crowdin.mobile-sdk-android:sdk:1.4.1') {
+   implementation ('com.github.crowdin.mobile-sdk-android:sdk:1.4.3') {
```
Comment (andrii-bodnar) [inline README.md:79]: The same here

File: crowdin/build.gradle [hunk 3]
```diff
@@ -1,11 +1,25 @@
...
+    id 'kotlin-android'
+}
+
 apply from: '../jacoco.gradle'
 
 ext {
-    libraryVersionCode = 17
-    libraryVersionName = "1.4.1"
+    libraryVersionCode = 19
+    libraryVersionName = "1.4.3"
```
Comment (andrii-bodnar) [inline crowdin/build.gradle:22]: The same here

File: README.md (hunk 1 above)
Comment (MykhailoNester) [reply README.md]: updated to 1.4.2. Was changed by mistake in two commits.

**Review by andrii-bodnar — APPROVED** [review]

File: build.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 buildscript {
 
     ext {
-        kotlin_version = '1.4.10'
+        kotlin_version = '1.4.31'
```
Comment (Atom1985) [inline build.gradle:4]: build.gradle

**Review by Atom1985 — COMMENTED** [review]
0e14cdf1d7a3a1ef85b89b64191042cd8e006ecd

**Pull request merged**